File: leetcode/101-200/172-factorial-trailing-zeroes/bench/trailing_zeroes.py
"""Benchmark workload for LeetCode #172 — Factorial Trailing Zeroes (Python; scale lane)."""


def trailing_zeroes(n):
    count = 0
    m = n // 5
    while m > 0:
        count += m
        m //= 5
    return count


def main():
    limit = 35000000
    sink = 0
    for i in range(limit):
        sink += trailing_zeroes(i)
    print(sink)


main()

File: leetcode/101-200/172-factorial-trailing-zeroes/bench/trailing_zeroes.rs
fn trailing_zeroes(n: i64) -> i64 {
    let mut count = 0i64;
    let mut m = n / 5;
    while m > 0 {
        count += m;
        m /= 5;
    }
    count
}

fn main() {
    let limit: i64 = 35000000;
    let mut sink: i64 = 0;
    for i in 0..limit {
        sink += trailing_zeroes(i);
    }
    println!("{}", sink);
}
